include/diy/master.hpp

```diff
@@ -703,6 +703,7 @@ diy::Master::
 foreach_(const Callback<Block>& f, const Skip& skip)
 {
     auto scoped = prof.scoped("foreach");
+    (void)scoped;
```

Comment:
For these kinds of unused variables, let's add a macro `DIY_UNUSED` and label them with it. Here's a version of it from `examples/simple/simple.cpp`:
```
#define UNUSED(expr) do { (void)(expr); } while (0)
```